pkg/kubectl/cmd/util/printing.go

```diff
@@ -81,7 +83,10 @@ func OutputVersion(cmd *cobra.Command, defaultVersion string) string {
 // Requires that printer flags have been added to cmd (see AddPrinterFlags).
 func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error) {
 	outputFormat := GetFlagString(cmd, "output")
-	templateFile := GetFlagString(cmd, "template")
+	templateFile := GetFlagString(cmd, "golang-template")
```

Comment:
I tend to 3 as well, although the names are pretty long to type :)

On Aug 6, 2015, at 1:18 AM, Brian Grant <notifications@github.com> wrote:

In pkg/kubectl/cmd/util/printing.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12220#discussion_r36381964>
:

> @@ -81,7 +83,10 @@ func OutputVersion(cmd *cobra.Command, defaultVersion string) string {
>  // Requires that printer flags have been added to cmd (see AddPrinterFlags).
>  func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error) {
>  	outputFormat := GetFlagString(cmd, "output")
> -	templateFile := GetFlagString(cmd, "template")
> +	templateFile := GetFlagString(cmd, "golang-template")

Alternatives:
1. -o template and different flags for each template format:
--golang-template=..., --jsonpath-template=..., etc.
2. -o golang-template, -o jsonpath-template, etc. with one template flag `
--template=....
3. -o golang-template=..., -o jsonpath-template=..., etc.

I think @thockin <https://github.com/thockin> is in favor of (3). It does
seem simpler.

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/12220/files#r36381964>
.